hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/MemberPruningTest.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.cluster.Member;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.PartitioningAttributeConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.cluster.MemberInfo;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.DAG;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.impl.JobInvocationObserver;
+import com.hazelcast.jet.sql.SqlTestSupport;
+import com.hazelcast.partition.Partition;
+import com.hazelcast.partition.PartitionService;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toSet;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class MemberPruningTest extends SqlTestSupport {
+
+    public static final int MEMBER_COUNT = 5;
+    private static final JobInvocationObserverImpl JOB_INVOCATION_OBSERVER = new JobInvocationObserverImpl();
```

Comment:
`SqlEndToEndTestSupport` registers new `jobInvocationObserver` instance for each test method, but does not unregister previous ones. This is not harmful, but could be improved. Here you share the observer between tests, which might make them dependent